File: nhglib-core/src/io/github/movementspeed/nhglib/utils/data/Strings.java
package io.github.movementspeed.nhglib.utils.data;

/**
 * Created by Fausto Napoli on 25/10/2016.
 */
public class Strings {
    public static class Messages {
        public final static String cannotQueueAssetFileNotFound = "Cannot queue asset \"%s\". File does not exist.";
        public final static String assetLoaded = "Asset \"%s\" has been loaded.";
        public final static String nullAssetSource = "Asset source cannot be null.";
        public final static String nullOrEmptyAssetSource = "Asset source cannot be null or empty.";
        public final static String controllerConnected = "Controller %s connected.";
        public final static String controllerDisconnected = "Controller %s disconnected.";
    }

    public static class Defaults {
        public final static String assetKey = "nhg_asset";
        public final static String assetPackageKey = "nhg_asset_package";
        public final static String inputKey = "nhg_input";
        public final static String sceneKey = "nhg_scene";
    }

    public static class Events {
        public final static String engineDestroy = "nhg_event_engine_destroy";
        public final static String assetLoadingFinished = "nhg_event_asset_loading_finished";
        public final static String assetLoaded = "nhg_event_asset_loaded";
        public final static String assetPackageLoaded = "nhg_event_asset_package_loaded";
        public final static String assetUnloaded = "nhg_event_asset_unloaded";
        public final static String keyInputTriggered = "nhg_event_input_triggered";
        public final static String sceneLoaded = "nhg_event_scene_loaded";
    }

    public static class RenderingSettings {
        public final static String forceUnlitKey = "nhg_force_unlit";
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/utils/data/Strings.kt
package io.github.movementspeed.nhglib.utils.data

/**
 * Created by Fausto Napoli on 25/10/2016.
 */
class Strings {
    object Messages {
        const val cannotQueueAssetFileNotFound = "Cannot queue asset \"%s\". File does not exist."
        const val assetLoaded = "Asset \"%s\" has been loaded."
        const val nullAssetSource = "Asset source cannot be null."
        const val nullOrEmptyAssetSource = "Asset source cannot be null or empty."
        const val controllerConnected = "Controller %s connected."
        const val controllerDisconnected = "Controller %s disconnected."
    }

    object Defaults {
        const val assetKey = "nhg_asset"
        const val assetPackageKey = "nhg_asset_package"
        const val inputKey = "nhg_input"
        const val sceneKey = "nhg_scene"
    }

    object Events {
        const val engineDestroy = "nhg_event_engine_destroy"
        const val enginePause = "nhg_event_engine_pause"
        const val engineResume = "nhg_event_engine_resume"
        const val assetLoadingFinished = "nhg_event_asset_loading_finished"
        const val assetLoaded = "nhg_event_asset_loaded"
        const val assetPackageLoaded = "nhg_event_asset_package_loaded"
        const val assetUnloaded = "nhg_event_asset_unloaded"
        const val keyInputTriggered = "nhg_event_input_triggered"
        const val sceneLoaded = "nhg_event_scene_loaded"
    }

    object RenderingSettings {
        const val forceUnlitKey = "nhg_force_unlit"
    }
}
